app/src/main/java/com/practicum/playlistmaker/SearchHistory.kt

```diff
@@ -0,0 +1,51 @@
+package com.practicum.playlistmaker
+
+import android.content.SharedPreferences
+import com.google.gson.Gson
+
+class SearchHistory(private val tracksInHistory: ArrayList<Track>) {
+
+    fun getTracksFromSharedPrefs(sharedPrefs: SharedPreferences): ArrayList<Track> {
+
+        val historyInSharedPrefs = sharedPrefs.getString(HISTORY_SEARCH_KEY, null)
+
+        if(historyInSharedPrefs != null) {
+            val tracksFromSharedPrefs = Gson().fromJson(historyInSharedPrefs, Array<Track>::class.java)
+            for(track in tracksFromSharedPrefs) {
+                tracksInHistory.add(track)
+            }
+        }
+        return tracksInHistory
+    }
+
+    fun putTracksInSharedPrefs(sharedPrefs: SharedPreferences) {
+        sharedPrefs.edit()
+            .putString(HISTORY_SEARCH_KEY, Gson().toJson(tracksInHistory))
+            .apply()
+    }
+
+    fun addInHistory(track: Track): ArrayList<Track> {
+
+        val iterator = tracksInHistory.iterator()
+        while (iterator.hasNext()) {
+            val nextTrack = iterator.next()
+            if(nextTrack.trackId == track.trackId) {
+                iterator.remove()
+            }
+        }
```

Comment:
🍏 Можно использовать специальный метод удаления из коллекции, который принимает в себя условие:
```suggestion
        tracksInHistory.removeIf { it.trackId == track.trackId }
```